libs/agno/agno/knowledge/knowledge.py

```diff
@@ -450,75 +466,57 @@ async def _load_from_url(
             self._update_content(content)
             log_warning(f"Invalid URL: {content.url} - {str(e)}")
 
-        # Determine file type from URL
-        url_path = Path(parsed_url.path)  # type: ignore
-        file_extension = url_path.suffix.lower()
-        read_documents = []
-        try:
-            if content.url.endswith("llms-full.txt") or content.url.endswith("llms.txt"):  # type: ignore
-                log_info("Detected llms, using url reader")
-                reader = content.reader or self.url_reader
-                if reader is not None:
-                    # TODO: We will refactor this to eventually pass authorization to all readers
-                    import inspect
-
-                    read_signature = inspect.signature(reader.read)
-                    if "password" in read_signature.parameters and content.auth and content.auth.password:
-                        read_documents = reader.read(content.url, name=content.name, password=content.auth.password)
-                    else:
-                        read_documents = reader.read(content.url, name=content.name)
-
-            elif file_extension and file_extension is not None:
-                log_info(f"Detected file type: {file_extension} from URL: {content.url}")
-                if content.reader:
-                    reader = content.reader
-                else:
-                    reader = self._select_url_file_reader(file_extension)
-                if reader is not None:
-                    log_info(f"Selected reader: {reader.__class__.__name__}")
-                    # TODO: We will refactor this to eventually pass authorization to all readers
-                    import inspect
+        # 3. Fetch and load content
+        async with AsyncClient() as client:
+            response = await async_fetch_with_retry(content.url, client=client)
+        bytes_content = BytesIO(response.content)
 
-                    read_signature = inspect.signature(reader.read)
-                    if "password" in read_signature.parameters and content.auth and content.auth.password:
-                        read_documents = reader.read(content.url, name=content.name, password=content.auth.password)
-                    else:
-                        read_documents = reader.read(content.url, name=content.name)
-                else:
-                    log_info(f"No reader found for file extension: {file_extension}")
+        # 4. Select reader
+        # If a reader was provided by the user, use it
+        reader = content.reader
+        # Else select based on file extension
+        if reader is None:
+            name = content.name
+            url_path = Path(parsed_url.path)
+            file_extension = url_path.suffix.lower()
+            if file_extension == ".csv":
+                name = basename(parsed_url.path) or "data.csv"
+                reader = self.csv_reader
+            elif file_extension == ".pdf":
```

Comment:
docx and the others? That is part of the win here, we now automatically support all formats